src/parser.rs

```diff
@@ -640,12 +659,75 @@ impl<'a, 'b> Parser<'a, 'b> {
+        let ident = Identifier::from(self.scanner.next().unwrap());
+
+        // TODO: loop this section and check for range expressions [i..j]
+        let peeked = self.scanner.peek()
+                         .expect("Expected an angle bracket or paren after an identifier.");
+         if let Token {
+                token_type: TokenType::LeftAngle,
+                ..
+            } = peeked {
+            // Have to perform manual port mappings for now
+            // Seperate the buffer mappings into a seperate function?
+             self.consume(TokenType::LeftAngle)?;
+
+             let peeked1 = self.scanner.peek()
+                               .expect("Expected and equals sign or a generic declaration");
+             if let Token {
+                 token_type: TokenType::Equal,
+                 ..
+             } = peeked1 {
+                 self.consume(TokenType::Equal)?;
+                 let wire_ident = self.scanner.peek();
+                 self.consume(TokenType::Identifier)?;
+
+                 let mut port_mappings = Vec::<PortMapping>::new();
+                 port_mappings.push(PortMapping {
+                     wire_ident: ident.clone(),
+                     wire: BusHDL {
+                         name: ident.value.clone(),
+                         start: Some(GenericWidth::Terminal(Terminal::Num(0))),
+                         end: Some(GenericWidth::Terminal(Terminal::Num(0))),
+                     },
+                     port: BusHDL {
+                         name: String::from("out"),
+                         start: Some(GenericWidth::Terminal(Terminal::Num(0))),
+                         end: Some(GenericWidth::Terminal(Terminal::Num(0))),
+                     },
+                 });
+                 port_mappings.push(PortMapping {
+                     wire_ident: ident,
+                     wire: BusHDL {
+                         name: wire_ident.unwrap().lexeme,
+                         start: Some(GenericWidth::Terminal(Terminal::Num(0))),
+                         end: Some(GenericWidth::Terminal(Terminal::Num(0))),
+                     },
+                     port: BusHDL {
+                         name: String::from("in"),
+                         start: Some(GenericWidth::Terminal(Terminal::Num(0))),
+                         end: Some(GenericWidth::Terminal(Terminal::Num(0))),
+                     },
+                 });
+                 self.consume(TokenType::Semicolon)?;
+
+
+                 return Ok(Component {
+                     name: Identifier::from("buffer"),
+                     generic_params: Vec::new(), // no generic params for the buff chip
+                     mappings: port_mappings,
+                 });
+             }
+         }
+
+        return Ok(Component {
+            name: ident,
             generic_params: self.generics()?,
             mappings: self.port_mappings()?,
-        })
```

Comment:
We do want it to be the case that is someone actually types `Buffer(in=foo, out=blah);` that it becomes a buffer component at parse time. This would then be transformed into the hard-coded buffer implementation by the simulator.